src/Adapter/AmazonS3.php

```diff
@@ -337,8 +337,13 @@ private function pathOrUrlToPath($input)
              *  Try to detect the bucket name from the hostname
              */
             $host = parse_url($input, PHP_URL_HOST);
-            if (preg_match('/.s3[-\.a-z0-9]*\.amazonaws\.com$/', $host)) {
-                $bucket = substr($host, 0, strpos($host, '.s3'));
+            if (preg_match('/s3[-\.a-z0-9]*\.amazonaws\.com$/', $host)) {
```

Comment:
It would be good to insert come comments with examples of host names. To understand what is this change for